frontend/utilities/sql_tools.ts

```diff
@@ -73,46 +78,82 @@ const filterCompatiblePlatforms = (
 
 export const parseSqlTables = (
   sqlString: string,
-  includeCteTables = false
+  includeVirtualTables = false
 ): string[] => {
   let results: string[] = [];
 
-  // Tables defined via common table expression will be excluded from results by default
-  const cteTables: string[] = [];
+  // Tables defined via common table expression (WITH ... AS syntax) or as subselects
+  // will be excluded from results by default.
+  const virtualTables: string[] = [];
+
+  // Tables defined via functions like `json_each` will always be excluded from results.
+  const functionTables: string[] = [];
 
-  const _callback = (node: IAstNode) => {
+  const _callback = (node: IAstNode, parentKey: string) => {
     if (!node) {
       return;
     }
 
+    // Common Table Expressions (CTEs) using "WITH ... AS" syntax.
     if (
-      (node.variant === "common" || node.variant === "recursive") &&
-      node.format === "table" &&
-      node.type === "expression"
+      parentKey === "with" &&
```

Comment:
:eyes:

Reply:
The AST produced by the new parser is completely different than `sql-parser`.  I'll upload a gist of what it looks like for the example query.

Reply:
https://gist.github.com/sgress454/3efafe2d673f4a581fc420c6d8d3505e